### PR TITLE
jrnl: support journal encryption by default

### PR DESCRIPTION
**Pull request by umazalakain**
###### Motivation for this change

Currently `jrnl` doesn't support encrypted journals because it doesn't depend on `pycrypto`. I enabled encryption by default, as I think it's a small dependency and the feature is a commonly used one.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by nixy** [comment]
If you are going to expose support `supportEncryption` as an option you will probably want to take it as an argument. I haven't done this before myself but I took a poke through python packages and found this example which does something similar to what you are trying to do.

```
  binwalk_fun = { visualizationSupport ? false, pyqtgraph ? null }:
    assert visualizationSupport -> pyqtgraph != null;

    buildPythonPackage rec {
    name = "binwalk-${version}";
    version = "2.1.1";

    src = pkgs.fetchFromGitHub {
      owner = "devttys0";
      repo = "binwalk";
      rev = "291a03595d17f848c73b74cb6ca508da782cd8f7";
      sha256 = "0grid93yz6i6jb2zggrqncp5awdf7qi88j5y2k7dq0k9r6b8zydw";
    };

    propagatedBuildInputs = with stdenv.lib; with pkgs; [ zlib xz ncompress gzip bzip2 gnutar p7zip cabextract lzma self.pycrypto ]
      ++ optional visualizationSupport pyqtgraph;

    meta = with stdenv.lib; {
      homepage = "http://binwalk.org";
      description = "A tool for searching a given binary image for embedded files";
      platforms = platforms.all;
      maintainers = [ maintainers.koral ];
    };
  };
```

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -11603,6 +11603,7 @@ in {
   jrnl = buildPythonPackage rec {
```
Comment (Mic92) [inline pkgs/top-level/python-packages.nix]: can you also move this package from  `python-packages.nix` to a dedicated file and add a reference to `all-packages.nix`. It is not a library and therefor does not belong here.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -11612,7 +11613,7 @@ in {
     propagatedBuildInputs = with self; [
       pytz six tzlocal keyring argparse dateutil_1_5
       parsedatetime
-    ];
+    ] ++ optional (supportEncryption) [ pycrypto ];
```
Comment (Mic92) [inline pkgs/top-level/python-packages.nix]: I think you can even just skip the option. It can be easily disabled by using `overrideDerivation`.

**Comment by unaizalakain** [comment]
I moved it into `pkgs/applications/misc` and chose to just drop the encryption support switch.

**Pull request closed without merging**